server/sonar-web/src/main/js/apps/projectActivity/components/ProjectActivityApp.js

```diff
@@ -20,79 +20,107 @@
 // @flow
 import React from 'react';
 import Helmet from 'react-helmet';
-import { connect } from 'react-redux';
+import moment from 'moment';
 import ProjectActivityPageHeader from './ProjectActivityPageHeader';
 import ProjectActivityAnalysesList from './ProjectActivityAnalysesList';
-import ProjectActivityPageFooter from './ProjectActivityPageFooter';
-import { fetchProjectActivity } from '../actions';
-import { getComponent } from '../../../store/rootReducer';
+import ProjectActivityGraphs from './ProjectActivityGraphs';
+import { GRAPHS_METRICS, activityQueryChanged } from '../utils';
 import { translate } from '../../../helpers/l10n';
 import './projectActivity.css';
+import type { Analysis, MeasureHistory, Metric, Query } from '../types';
 
 type Props = {
-  location: { query: { id: string } },
-  fetchProjectActivity: (project: string, filter: ?string) => void,
-  project: { configuration?: { showHistory: boolean } }
+  addCustomEvent: (analysis: string, name: string, category?: string) => Promise<*>,
+  addVersion: (analysis: string, version: string) => Promise<*>,
+  analyses: Array<Analysis>,
+  changeEvent: (event: string, name: string) => Promise<*>,
+  deleteAnalysis: (analysis: string) => Promise<*>,
+  deleteEvent: (analysis: string, event: string) => Promise<*>,
+  loading: boolean,
+  project: { configuration?: { showHistory: boolean }, key: string, leakPeriodDate: string },
+  metrics: Array<Metric>,
+  measuresHistory: Array<MeasureHistory>,
+  query: Query,
+  updateQuery: (newQuery: Query) => void
 };
 
 type State = {
-  filter: ?string
+  filteredAnalyses: Array<Analysis>
 };
 
-class ProjectActivityApp extends React.PureComponent {
+export default class ProjectActivityApp extends React.PureComponent {
   props: Props;
+  state: State;
 
-  state: State = {
-    filter: null
-  };
-
-  componentDidMount() {
-    const html = document.querySelector('html');
-    if (html) {
-      html.classList.add('dashboard-page');
-    }
-    this.props.fetchProjectActivity(this.props.location.query.id);
+  constructor(props: Props) {
+    super(props);
+    this.state = { filteredAnalyses: this.filterAnalyses(props.analyses, props.query) };
   }
 
-  componentWillUnmount() {
-    const html = document.querySelector('html');
-    if (html) {
-      html.classList.remove('dashboard-page');
+  componentDidUpdate(prevProps: Props) {
```

Comment:
do it in the `componentWillUpdate(nextProps)`, otherwise you have two renders: one for new props, one for new state